src/shared/Microsoft.Git.CredentialManager/Interop/Windows/WindowsSettings.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.Git.CredentialManager.Interop.Windows
+{
+    /// <summary>
+    /// Reads settings from Git configuration, environment variables, and defaults from the Windows Registry.
+    /// </summary>
+    public class WindowsSettings : Settings
+    {
+        private readonly ITrace _trace;
+
+        public WindowsSettings(IEnvironment environment, IGit git, ITrace trace)
+            : base(environment, git)
+        {
+            EnsureArgument.NotNull(trace, nameof(trace));
+            _trace = trace;
+
+            PlatformUtils.EnsureWindows();
+        }
+
+        protected override bool TryGetExternalDefault(string section, string property, out string value)
+        {
+            value = null;
+
+#if NETFRAMEWORK
+            // Check for machine (HKLM) registry keys that match the Git configuration name.
+            // These can be set by system administrators via Group Policy, so make useful defaults.
+            using (Win32.RegistryKey configKey = Win32.Registry.LocalMachine.OpenSubKey(Constants.WindowsRegistry.HKConfigurationPath))
```

Comment:
Why is this limited to `NETFRAMEWORK` projects? It looks like both `RegistryKey` and `Registry` exist for .NET 5.

Reply:
Although the `Registry*` types exist in .NET 5 (/.NET Core), they exist in a separate assembly/NuGet package that must be referenced to use. For the .NET Framework target the `Registry*` types are included by default in the framework references.

Since this feature only pertains to Windows, and we only release on .NET Framework for Windows there's no need to bring the extra assembly along for the ride in all the other targets (Mac and Linux).
If/once we move to .NET 5 (or 6) on Windows then there'll need to be a review of all `#if NETFRAMEWORK` code, at which point we should hopefully have better support from the linker/trimmer to keep distribution bloat to a minimum on other platforms.